File: internal/domain/bot.go
package domain

type BotStatus string

const (
	BotStatusActive BotStatus = "ACTIVE"
	BotStatusIdle   BotStatus = "IDLE"
)

type Bot struct {
	ID     uint
	Status BotStatus
	Order  *Order
}

func NewBot(id uint) *Bot {
	return &Bot{
		ID:     id,
		Status: BotStatusActive,
	}
}

func (b *Bot) Idle() {
	b.Status = BotStatusIdle
}

File: src/models/bot.js
const { BOT_STATUS } = require('../constants/statuses');

class Bot {
  constructor(id) {
    this.id = id;
    this.currentOrder = null;
    this.timer = null;
  }

  get isIdle() {
    return this.currentOrder === null;
  }

  get status() {
    return this.isIdle ? BOT_STATUS.IDLE : BOT_STATUS.ACTIVE;
  }
}

module.exports = Bot;
